watertap/unit_models/tests/test_thickener_unit.py

```diff
@@ -0,0 +1,279 @@
+
+        iscale.calculate_scaling_factors(tu)
+        initialization_tester(tu)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, tu):
+        solver = get_solver()
+        results = solver.solve(tu)
+        assert_optimal_termination(results)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solution(self, tu):
+        assert pytest.approx(101325.0, rel=1e-3) == value(
+            tu.fs.unit.overflow.pressure[0]
+        )
+        assert pytest.approx(308.15, rel=1e-3) == value(
+            tu.fs.unit.overflow.temperature[0]
+        )
+        assert pytest.approx(0.003115, rel=1e-3) == value(
+            tu.fs.unit.overflow.flow_vol[0]
+        )
+        assert pytest.approx(0.02806, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "S_I"]
+        )
+        assert pytest.approx(0.000673, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "S_S"]
+        )
+        assert pytest.approx(0.06768, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "X_I"]
+        )
+        assert pytest.approx(0.001409, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "X_S"]
+        )
+        assert pytest.approx(0.04286, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "X_P"]
+        )
+        assert pytest.approx(0.099046, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "X_BH"]
+        )
+        assert pytest.approx(0.007414, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "X_BA"]
+        )
+        assert pytest.approx(0.001374, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "S_O"]
+        )
+        assert pytest.approx(0.009194, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "S_NO"]
+        )
+        assert pytest.approx(0.0001584, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "S_NH"]
+        )
+        assert pytest.approx(0.0005594, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "S_ND"]
+        )
+        assert pytest.approx(0.0001056, rel=1e-3) == value(
+            tu.fs.unit.overflow.conc_mass_comp[0, "X_ND"]
+        )
+        assert pytest.approx(0.004564, rel=1e-3) == value(
+            tu.fs.unit.overflow.alkalinity[0]
+        )
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_conservation(self, tu):
+        assert (
+            abs(
+                value(
+                    tu.fs.unit.inlet.flow_vol[0] * tu.fs.props.dens_mass
+                    - tu.fs.unit.overflow.flow_vol[0] * tu.fs.props.dens_mass
+                    - tu.fs.unit.underflow.flow_vol[0] * tu.fs.props.dens_mass
```

Comment:
should we also look at Q*C balances? In reality, the underflow density would be different than the overflow although I don't know if we are given any expressions to approximate anything better than a single assumed solution density equal to that of water.

Reply:
I agree, however the property package is built with the assumption that the density is the one ok water